sources/engine/Stride.Physics/Simulation.cs

```diff
@@ -1170,11 +1176,26 @@ public virtual void Recycle(ref Vector3 from, ref Vector3 to, CollisionFilterGro
                 Flags = 0;
                 CollisionFilterGroup = (int)filterGroup;
                 CollisionFilterMask = (int)filterMask;
+                HitNoContResp = hitNoContResp;
+            }
+
+            public override bool NeedsCollision(BulletSharp.BroadphaseProxy proxy0)
+            {
+                if (HitNoContResp == false 
+                    && proxy0.ClientObject is BulletSharp.CollisionObject co 
+                    && (co.CollisionFlags & BulletSharp.CollisionFlags.NoContactResponse) != 0)
+                {
+                    return false;
+                }
+
+                return base.NeedsCollision(proxy0);
             }
         }
 
         private abstract class StrideReusableConvexResultCallback : BulletSharp.ConvexResultCallback
         {
+            public bool HitNoContResp;
```

Comment:
Can we use a non-abbreviated name so that it's easier to understand?
If I understood correctly, this is the value of `hitTriggers` whose doc is "Whether this test should collide with <see cref="PhysicsTriggerComponentBase"/>". Maybe we can copy-paste this doc here too?